application/src/main/java/org/togetherjava/tjbot/features/roleapplication/ApplicationCreateCommand.java

```diff
@@ -0,0 +1,297 @@
+
+        generateRoleOptions(getData());
+        applicationApplyHandler = new ApplicationApplyHandler(roleApplicationSystemConfig);
+    }
+
+    /**
+     * Populates a {@link SlashCommandData} object with the proper arguments.
+     *
+     * @param data the object to populate
+     */
+    private void generateRoleOptions(SlashCommandData data) {
+        IntStream.range(1, OPTIONAL_ROLES_AMOUNT + 1).forEach(index -> {
+            data.addOption(OptionType.STRING, generateOptionId("title", index),
+                    "The title of the role");
+            data.addOption(OptionType.STRING, generateOptionId("description", index),
+                    "The description of the role");
+            data.addOption(OptionType.STRING, generateOptionId("emoji", index),
+                    "The emoji of the role");
+        });
+    }
+
+    private static String generateOptionId(String name, int id) {
+        return "%s%s%d".formatted(name, VALUE_DELIMITER, id);
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        if (!handleHasPermissions(event)) {
+            return;
+        }
+
+        final List<OptionMapping> optionMappings = event.getInteraction().getOptions();
+        if (optionMappings.isEmpty()) {
+            event.reply("You have to select at least one role.").setEphemeral(true).queue();
+            return;
+        }
+
+        long incorrectArgsCount = getIncorrectRoleArgsCount(optionMappings);
+        if (incorrectArgsCount > 0) {
+            event.reply("Missing information for %d roles.".formatted(incorrectArgsCount))
+                .setEphemeral(true)
+                .queue();
+            return;
+        }
+
+        sendMenu(event);
+    }
+
+    @Override
+    public void onStringSelectSelection(StringSelectInteractionEvent event, List<String> args) {
+        SelectOption selectOption = event.getSelectedOptions().getFirst();
+
+        if (selectOption == null) {
+            return;
+        }
+
+        OffsetDateTime timeSentCache = applicationApplyHandler.getApplicationSubmitCooldown()
+            .getIfPresent(event.getMember());
+        if (timeSentCache != null) {
+            Duration duration = Duration.between(timeSentCache, OffsetDateTime.now());
+            long remainingMinutes =
+                    roleApplicationSystemConfig.applicationSubmitCooldownMinutes() - duration.toMinutes();
+
+            if (duration.toMinutes() < roleApplicationSystemConfig.applicationSubmitCooldownMinutes()) {
+                event
+                    .reply("Please wait %d minutes before sending a new application form."
+                        .formatted(remainingMinutes))
+                    .setEphemeral(true)
+                    .queue();
+                return;
+            }
+        }
+
+        String questionLabel = roleApplicationSystemConfig.defaultQuestion();
+        if (questionLabel.length() > TextInput.MAX_LABEL_LENGTH) {
```

Comment:
We can move this entire check into the record. Up to you, not sure if we'd want an exception on start or a concatenated message.